### ISSUE TITLE
[Bug] Preferences seem really finicky

### ISSUE THREAD

**Issue by remyabel2** [iv-org/invidious#2854]
**Describe the bug**
Preferences do not seem to be persistent and reset at random. The cookies oddly enough stay persistent, but the settings do not. Note that I clear cookies on shutdown but have added yewtu.be as an exception. The same behavior is exhibited with another instance. This process works for other sites, so I think the problem is with Invidious.

**Steps to Reproduce**

1. Go to https://yewtu.be/
2. Click on the preferences gear icon
3. Set a preference, such as "Save video playback position"
4. Navigate away from the page and back to the gear icon or closing the browser
5. The cookies should still be there, but the settings sometimes do not persist
6. Clearing the cookies seem to allow it to work again, but it will break again

When it's "broken", the preferences do not seem to save at all.

**Logs**

**Screenshots**

![Screenshot_2022-02-02_06-56-20](https://user-images.githubusercontent.com/94518122/152149244-d8a495fc-3151-4a23-bd9f-692efda57530.png)
![Screenshot_2022-02-02_06-59-07](https://user-images.githubusercontent.com/94518122/152149563-78a2a1d8-ff1a-49af-90ef-e0e26cf42fe0.png)

**Additional context**

Browser: Firefox 96

**Comment by unixfox** [comment]
Did you try with other browsers? I really can't reproduce your issue.

**Comment by remyabel2** [comment]
> Did you try with other browsers? I really can't reproduce your issue.

I don't have access to another browser. I tested it in a clean profile and didn't reproduce the issue. In this profile I'm using arkenfox which has some privacy settings. However with that being said, I still think the way Invidious sets cookies is probably the culprit.

I can see here that the request cookie (when it's working properly) is populated:

![Screenshot_2022-02-02_09-40-14](https://user-images.githubusercontent.com/94518122/152175363-01b9b503-e7f6-47e6-9735-af2fcd5ff245.png)

And firefox produces this diagnostic:

> Cookie “PREFS” has “SameSite” policy set to “Lax” because it is missing a “SameSite” attribute, and “SameSite=Lax” is the default value for this attribute.

And points to this line:

> document.cookie = `PREFS=${encodeURIComponent(JSON.stringify(prefs))}`;

When I restart my browser, the cookie is still there, but the request no longer contain the PREFS cookie. 

I also noticed that for some reason the domain for the cookie is `.yewtu.be` not `yewtu.be`. I'm not sure if this is enough for Firefox to consider it a "cross-origin" cookie.

While I'm going to troubleshoot my Firefox profile, I don't get this issue with other sites, so I'd still like to know what Invidious is doing differently.

**Comment by SamantazFox** [comment]
@remyabel2 does it happen on other invidious instances?
See https://api.invidious.io/ for a list of instances to try that on.

**Comment by remyabel2** [comment]
Yeah I tested it on another instance before opening this issue. Just tested it on a third instance and the issue wasn't there. Very weird. Guess this is just an instance issue.

**Comment by unixfox** [comment]
> Yeah I tested it on another instance before opening this issue. Just tested it on a third instance and the issue wasn't there. Very weird. Guess this is just an instance issue.

That's very strange, could you send us the instance where it worked?

**Comment by remyabel2** [comment]
The one that worked was https://invidious-us.kavin.rocks/.

Both yewtu.be and https://vid.puffyan.us/ exhibited the same behavior in the exact same way with the steps I took. puffyan's version is 2022.01.25-2ae074a9 but kavin's version is 2022.02.02-0ca3337, I do not think it's directly related to the invidious code anymore, maybe there's something going on with how it's deployed.

**Comment by remyabel2** [comment]
Okay, I'm an idiot guys. Turns out that the issue has nothing to do with my config settings or anything of that nature.

The script I pointed to is `persist-invidious-prefs.js` which I was not able to find in the source code or the patches for yewtu.be. The URL however is a moz extension. It then immediately clicked that this feature is part of the Privacy Redirect addon which has the ability to "persist invidious preferences". So in other words the issue has nothing to do with invidious at all. Sorry for the noise!
